akka-stream-tests/src/test/scala/akka/stream/scaladsl/SealedAttributesSpec.scala

```diff
@@ -0,0 +1,268 @@
+    //      )(Keep.right)
+    //
+    //      val maybeAttr = attributesOf(it)
+    //
+    //      info("initialAttributes.get[Name]: " + maybeAttr)
+    //      maybeAttr.get should equal(Attributes.name("initial-inside").attributeList.head)
+    //    }
+    //
+    //    "getInside; initial attrs -- override on stage" in {
+    //      val it = Source.empty.toMat(
+    //        Sink.fromGraph(
+    //          graphStageSinkShape(_.get[Name], initialAttr = Attributes.name("initial-inside"))
+    //        ).named("on-sink-override")
+    //      )(Keep.right)
+    //
+    //      val maybeAttr = attributesOf(it)
+    //
+    //      info("initialAttributes.get[Name]: " + maybeAttr)
+    //      maybeAttr.get should equal(Attributes.name("initial-inside").attributeList.head)
+    //    }
+    //
+    //    "getInside; initial attrs -- override on composite ('complete') stage" in {
+    //      val it = Source.empty.toMat(
+    //        Sink.fromGraph(
+    //          graphStageSinkShape(_.get[Name], initialAttr = Attributes.name("initial-inside"))
+    //        ).named("on-sink-override")
+    //      )(Keep.right).named("on-sink-outside-override")
+    //
+    //      //  >>> inheritedAttributes = Attributes(List(Name(initial-inside), Name(on-sink-override), Name(on-sink-outside-override), Name(on-sink-outside-override), Dispatcher(akka.test.stream-dispatcher), InputBuffer(2,16), SupervisionStrategy(<function1>)))
+    //      // >>> names = initial-inside nested in: on-sink-override nested in: on-sink-outside-override nested in: on-sink-outside-override
+    //      val maybeAttr = attributesOf(it)
+    //
+    //      info("initialAttributes.get[Name]: " + maybeAttr)
+    //      maybeAttr.get should equal(Attributes.name("initial-inside").attributeList.head)
+    //    }
+
+    // new semantics:
+
+    "xoxo getInside; dispatcher, override, no seal; initial attrs -- override on composite ('complete') stage" in {
+
+      /*
+          Here wa have a dispatcher, but we allow it to be overriden
+       */
+
+      val it = Source.empty.toMat(
+        Sink.fromGraph(
+          graphStageSinkShape(_.get[Dispatcher], initialAttr = dispatcher("dispatcher-default-in-stage-initial"))
+        ).withAttributes(dispatcher("right-on-sink"))
+      )(Keep.right)
+
+      val maybeAttr = attributesOf(it)
+
+      info("initialAttributes.get[Dispatcher]: " + maybeAttr)
+      maybeAttr.get.dispatcher should equal("right-on-sink")
+    }
+
+    "zozo getInside; dispatcher, override, seal; initial attrs -- override on composite ('complete') stage" in {
+
+      /*
+          Here wa have a dispatcher, we don't allow overrides
+       */
+
+      val it = Source.empty.toMat(
+        Sink.fromGraph(
+          graphStageSinkShape(_.get[Dispatcher], initialAttr = dispatcher("dispatcher-default-in-stage-initial").seal())
+        ).withAttributes(dispatcher("right-on-sink"))
+      )(Keep.right)
+
+      val maybeAttr = attributesOf(it)
```

Comment:
Note that:

` >>> inheritedAttributes = Attributes(List(Dispatcher(dispatcher-default-in-stage-initial), Dispatcher(right-on-sink), Dispatcher(akka.test.stream-dispatcher), InputBuffer(2,16), SupervisionStrategy(<function1>)))`